pkg/suggest/rules.go

```diff
@@ -0,0 +1,179 @@
+package suggest
+
+import (
+	"errors"
+	"fmt"
+	"github.com/agext/levenshtein"
+	"regexp"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+	"strings"
+)
+
+// NewStrReplaceRule creates a Rule that finds and replaces a string in the error message
+func NewStrReplaceRule(find, replace string) Rule {
```

Comment:
Do we need to export all the functions? I think the only one used outside the package is `UserFriendlyError`

Reply:
yes, I've refactored having `UserFriendlyError` so I will adjust before opening the PR :)